xdsl/dialects/arm/register.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from xdsl.backend.register_type import RegisterType
+from xdsl.ir import Attribute
+from xdsl.irdl import irdl_attr_definition
+from xdsl.parser import AttrParser
+from xdsl.utils.exceptions import VerifyException
+
+
+class ARMRegisterType(RegisterType):
+    """
+    The abstract class for all ARM register types.
+    """
+
+    @classmethod
+    def parse_parameters(cls, parser: AttrParser) -> Sequence[Attribute]:
+        if parser.parse_optional_punctuation("<") is not None:
+            name = parser.parse_identifier()
+            parser.parse_punctuation(">")
+            if not name.startswith("x"):  # only including x regs for now
+                assert (
+                    name in cls.abi_index_by_name()
+                ), f"{name}"  # abi_index_by_name raises NotImplementedError (?)
```

Comment:
this bit of code doesn't make sense right now, as all the registers below do start with x, so the assert will never trigger.

Reply:
I would say let's not verify any of this for the time being, and allow any string in the register name, we can always add the verification later.